### ISSUE TITLE
train  trick some questions

### ISSUE THREAD

**Issue by Mr-Monday** [KunpengLi1994/VSRN#24]
Hi,I am learning your VSRN code. I have some questions.  
When training and testing, does this " self.caption_model = S2VTAttModel(self.encoder, self.decoder) " code need to  be changed to  self.caption_model.train() and self.caption_model.eval() in def train_start(self) and def val_start(self). as follows 

    def train_start(self):
        """switch to train mode
        """
        self.img_enc.train()
        self.txt_enc.train()
       self.caption_model.train() 

    def val_start(self):
        """switch to evaluate mode
        """
        self.img_enc.eval()
        self.txt_enc.eval()
        self.caption_model.eval()

**Comment by KunpengLi1994** [comment]
Hi Thanks for your interest in our work. Since caption_model is only used during training to enhance the visual representation learning and it is not used for testing, it is not necessary to add self.caption_model.eval() to "def val_start(self)". As for training, if I remember correctly, the default setting of the module is .train(), therefore, I think it is also not necessary to add this line to "def train_start(self)".

**Comment by Mr-Monday** [comment]
Thank you for your reply，I found out later when I was reading the code that The model did not use this module in the test. There is really no need to add this module this line to "def train_start(self)".

**Comment by Mr-Monday** [comment]
Hi，When I review this code, I found in the "train.py" not used "def val_start(self) ",and I can not find where used the val_start(self) when validate. So when validate, do I need to used this function in def validate(opt, val_loader, model)? as follows：
def validate(opt, val_loader, model):
        model.val_start()
        img_embs, cap_embs = encode_data(
        model, val_loader, opt.log_step, logging.info)
       ...
